pandas/_libs/lib.pyx

```diff
@@ -1284,13 +1284,12 @@ cdef class Seen:
 
     @property
     def is_bool(self):
-        return not (self.datetime_ or self.numeric_ or self.timedelta_
-                    or self.nat_)
-
-    @property
-    def is_float_or_complex(self):
-        return not (self.bool_ or self.datetime_ or self.timedelta_
-                    or self.nat_)
+        # i.e. not (anything but bool)
```

Comment:
Would be nice if this was `def anything_but_bool` as it's odd that `self.bool_ = False` would still return `True`. Maybe a followup